org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/DetectVMInstallationsJob.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.IVMInstallType;
+import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.launching.VMStandin;
+import org.eclipse.osgi.util.NLS;
+
+class DetectVMInstallationsJob extends Job {
+
+	private static final Object FAMILY = "Family " + DetectVMInstallationsJob.class.getName(); //$NON-NLS-1$
+	
+	DetectVMInstallationsJob() {
+		super(LaunchingMessages.lookupInstalledJVMs);
+	}
+
+	@Override
+	protected IStatus run(IProgressMonitor monitor) {
+		Collection<File> rootDirectories = new HashSet<>();
+		if (!Platform.OS_WIN32.equals(Platform.getOS())) {
+			rootDirectories.add(new File("/usr/lib/jvm")); //$NON-NLS-1$
+		}
+		rootDirectories.add(new File(System.getProperty("user.home"), ".sdkman/candidates/java")); //$NON-NLS-1$ //$NON-NLS-2$
+		String javaHome = System.getenv("JAVA_HOME"); //$NON-NLS-1$
+		if (javaHome != null) {
+			rootDirectories.add(new File(javaHome));
+		}
+		String jdkHome = System.getenv("JDK_HOME"); //$NON-NLS-1$
+		if (jdkHome != null) {
+			rootDirectories.add(new File(jdkHome));
+		}
+		// other common/standard lookup strategies can be added here
+
+		Collection<File> candidates = rootDirectories.stream().filter(File::isDirectory)
+			.map(dir -> dir.listFiles(File::isDirectory))
+			.filter(Objects::nonNull)
+			.flatMap(Arrays::stream)
+			.map(t -> {
+				try {
+					return t.getCanonicalFile();
+				} catch (IOException e) {
+					return null;
+				}
+			}).filter(Objects::nonNull)
+			.collect(Collectors.toCollection(HashSet::new));
+		if (monitor.isCanceled()) {
+			return Status.CANCEL_STATUS;
+		}
+		Set<File> knownInstallations = Stream.of(JavaRuntime.getVMInstallTypes())
+			.map(IVMInstallType::getVMInstalls)
+			.flatMap(Arrays::stream)
+			.map(IVMInstall::getInstallLocation)
+			.filter(Objects::nonNull)
+			.map(t -> {
+				try {
+					return t.getCanonicalFile();
+				} catch (IOException e) {
+					return null;
+				}
+			}).filter(Objects::nonNull)
+			.collect(Collectors.toSet());
+		candidates.removeIf(knownInstallations::contains);
+		StandardVMType standardType = (StandardVMType)JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+		candidates.forEach(f -> {
+			if (monitor.isCanceled()) {
+				return;
+			}
+			new Job(NLS.bind(LaunchingMessages.configuringJVM, f.getAbsolutePath())) {
+				@Override
+				public IStatus run(IProgressMonitor monitor) {
+					VMStandin workingCopy = new VMStandin(standardType, f.getAbsolutePath());
+					workingCopy.setInstallLocation(f);
+					workingCopy.setName(f.getName());
```

Comment:
`f.getName()` is surely not unique, so if any of candidates located at different root directories have same names, only the last (or the first?) will survive, others will cause errors or be ignored. I haven't checked where they are "added" to the list of "remembered" JRE's, but at least on the UI it is not possible to add new JRE that has same name as one of already defined.

Reply:
OK. Should we use the absolute path as name then?

Reply:
I verified and the check in UI is not an API requirement. I could install multiple JVMs with same name
![Screenshot from 2023-04-17 17-02-19](https://user-images.githubusercontent.com/469058/232527198-f2f45bd2-4d88-4c51-ad07-858db96c648a.png)
 (via API with this code, copying JVM installations to JDK_HOME)

Reply:
OK, I've changed it and copied the logic from the "Search" button to suffix with `(1)`, `(2)`,...